test/jdk/javax/swing/JButton/bug4490179.java

```diff
@@ -53,10 +56,8 @@ public static void main(String[] args) throws Exception {
                 frame = new JFrame("bug4490179");
```

Comment:
Instead of three volatile fields — `pt`, `buttonW`, `buttonH` — there could be only one:

```java
            SwingUtilities.invokeAndWait(() -> {
                Point loc = button.getLocationOnScreen();
                Dimension size = button.getSize(); 
                pt = new Point(loc.x + size.width / 2,
                               loc.y + size.height / 2);
            });
```

I think this better conveys the idea, less variables / fields to track in your mind.